RELEASE/scripts/autoscend/auto_util.ash

```diff
@@ -2100,6 +2101,81 @@ boolean acquireGumItem(item it)
 	return (have + 1) == item_amount(it);
 }
 
+boolean acquireTotem()
+{
+	//this function checks if you have a valid totem for casting turtle tamer buffs with. Returning true if you do. If you don't, it will attempt to acquire one in a reasonable manner.
+
+	//first check if there is a valid totem in inventory or equipped, return true if there is.
+	
+	if ((0 < item_amount($item[primitive alien totem])) || (have_equipped($item[primitive alien totem])))
+	{
+		return true;
+	}
+	if ((0 < item_amount($item[Flail of the Seven Aspects])) || (have_equipped($item[Flail of the Seven Aspects])))
+	{
+		return true;
+	}
+	if ((0 < item_amount($item[Chelonian Morningstar])) || (have_equipped($item[Chelonian Morningstar])))
+	{
+		return true;
+	}
+	if ((0 < item_amount($item[Mace of the Tortoise])) || (have_equipped($item[Mace of the Tortoise])))
+	{
+		return true;
+	}
+	if ((0 < item_amount($item[Ouija Board, Ouija Board])) || (have_equipped($item[Ouija Board, Ouija Board])))
+	{
+		return true;
+	}
+	if ((0 < item_amount($item[turtle totem])) || (have_equipped($item[turtle totem])))
+	{
+		return true;
+	}
+	
+	//if no valid totems were found, check the closet for one, from best to worst. If found in closet, uncloset 1 and return true
+	
+	if (0 < closet_amount( $item[primitive alien totem] ))
+	{
+		take_closet( 1 , $item[primitive alien totem] );
+		return true;
+	}
+	if (0 < closet_amount( $item[Flail of the Seven Aspects] ))
+	{
+		take_closet( 1 , $item[turtle totem] );
+		return true;
+	}
+	if (0 < closet_amount( $item[Chelonian Morningstar] ))
+	{
+		take_closet( 1 , $item[turtle totem] );
+		return true;
+	}
+	if (0 < closet_amount( $item[Mace of the Tortoise] ))
+	{
+		take_closet( 1 , $item[turtle totem] );
+		return true;
+	}
+	if (0 < closet_amount( $item[Ouija Board, Ouija Board] ))
+	{
+		take_closet( 1 , $item[turtle totem] );
+		return true;
+	}
+	if (0 < closet_amount( $item[turtle totem] ))
+	{
+		take_closet( 1 , $item[turtle totem] );
+		return true;
+	}
```

Comment:
as above.

Reply:
Does foreach process the items from right to left or left to right?
it's an edge case, but if you have closeted more than item they should be checked in order from best to worst.

Reply:
[Doc's](https://wiki.kolmafia.us/index.php?title=Data_Types#Plural_Typed_Constants) don't say but I'd assume left to right and you're correct you probably want to check for the best first in this case.

I'll test which order plurals are processed in when I'm at home this evening just to be sure but I'd be very surprised if it was anything other than left to right.

Reply:
I just tested it with
```
foreach totem in $items[turtle totem, Mace of the Tortoise, Chelonian Morningstar, Flail of the Seven Aspects, Ouija Board\, Ouija Board, primitive alien totem]
{
	print(totem);
}
```
and it goes from left to right

Reply:
change made